File: tests/test08.py
#!/usr/bin/python2.7 -u
import re
# Test for subset4 - variable interpolation
# Taken from lecture slides
# http://www.cse.unsw.edu.au/~cs2041/13s2/lec/perl/perl.notes.html

pattern = "ab+"
replace = "Yod"
text = "abba"

text = re.sub(r'ab+', 'Yod', text)
print "%s" % (text)
# converts "abba" to "Yoda"

File: tests/test08.pl
#!/usr/bin/perl -w
# Test for subset4 - variable interpolation
# Taken from lecture slides
# http://www.cse.unsw.edu.au/~cs2041/13s2/lec/perl/perl.notes.html

$pattern = "ab+";
$replace = "Yod";
$text = "abba";

$text =~ s/$pattern/$replace/;
print "$text\n";
   # converts "abba" to "Yoda"
